src/mixins/chrome/chromeProvider/chromeProvider.ts

```diff
@@ -0,0 +1,54 @@
+import { html, HTMLTemplateResult, nothing, PropertyValues } from "lit";
+import { Component } from "../../mixins";
+import { ChromeAdvertisement, chromeAdvertisementEventName } from "../chromeHost/chromeHost";
+import { state } from "lit/decorators.js";
+import { AbstractComponent } from "../../abstractComponent";
+
+export type ChromeTemplate = HTMLTemplateResult | typeof nothing;
+
+export interface WithChromeProvider {
+  chromeTop?(): ChromeTemplate;
+  chromeBottom?(): ChromeTemplate;
+  chromeLeft?(): ChromeTemplate;
+  chromeRight?(): ChromeTemplate;
+  chromeOverlay?(): ChromeTemplate;
+}
+
+export type ChromeProviderKey = keyof WithChromeProvider;
+
+export const ChromeProvider = <T extends Component>(superClass: T) => {
+  abstract class ChromeProviderComponentClass extends superClass implements WithChromeProvider {
+    @state()
+    private chromeAdvertisement?: ChromeAdvertisement;
+
+    public abstract chromeTop?(): ChromeTemplate;
+    public abstract chromeBottom?(): ChromeTemplate;
+    public abstract chromeLeft?(): ChromeTemplate;
+    public abstract chromeRight?(): ChromeTemplate;
+    public abstract chromeOverlay?(): ChromeTemplate;
+
+    public firstUpdated(change: PropertyValues<this>): void {
+      super.firstUpdated(change);
+      this.addEventListener(chromeAdvertisementEventName, (event: any) => this.handleChromeAdvertisement(event));
+    }
+
+    public updated(change: PropertyValues<this>): void {
+      super.updated(change);
+    }
+
+    /** You might want to override this */
+    protected handleSlotChange(): void {}
```

Comment:
why might i want to override this?

Can we provide a useful default implementations?

Reply:
Actually, I don't need this anymore and is an artifact from testing.

`ChromeProvider` now extends `AbstractComponent` which has a `handleSlotChange` event.

I originally did this so that the `AbstractComponent` could override the `ChromeProvider` `handleSlotChange` method

Reply:
Update, my previous comment was incorrect.

The following comment explains why this method exists

![image](https://github.com/user-attachments/assets/db5c0342-d88e-42c8-9482-4a0d1e035677)

```ts
/**
 * A lifecycle method that is run when new elements are slotted in to the
 * providers default slot.
 *
 * This can be useful for attaching subscribers to nested components.
 * e.g. Subscribing to the spectrogram components unitConverter property
 * when the spectrogram is slotted into the provider.
 */
```